addons/account/models/account_tax.py

```diff
@@ -1104,10 +1105,19 @@ def prepare_tax_extra_data(tax, **kwargs):
                     base -= total_tax_amount
             tax_data['base'] = base
 
+            # Subsequent taxes.
+            tax_data['taxes'] = self.env['account.tax']
+            if tax.include_base_amount:
+                tax_data['taxes'] |= subsequent_taxes
+
             # Reverse charge.
             if tax.has_negative_factor:
                 reverse_charge_tax_data = reverse_charge_taxes_data[tax.id]
                 reverse_charge_tax_data['base'] = base
+                reverse_charge_tax_data['taxes'] = tax_data['taxes']
+
+            if tax.is_base_affected:
+                subsequent_taxes |= tax
```

Comment:
Note for the reviewer: The subsequent taxes was added by `_add_accounting_data_to_base_line_tax_details` before. However, it's convenient for the new helpers to know the taxes affected by the current one so I moved that part of the code in the taxes engine part